### PR TITLE
fix(draw): prevent zero scaleX/scaleY on draw and highlight shapes

### LINKED ISSUE #7997: ValidationError: At shape(type = draw).props.scaleY: Expected a non-zero number, got 0
Sentry Issue: [LITE-6AA](https://tldraw.sentry.io/issues/7170466418/?referrer=github_integration)

```
ValidationError: At shape(type = draw).props.scaleY: Expected a non-zero number, got 0
  at prefixError (../../../../../packages/validate/src/lib/validation.ts:174:10)
  at validateUsingKnownGoodVersionFn (../../../../../packages/validate/src/lib/validation.ts:1766:11)
  at Validator.validateUsingKnownGoodVersion (../../../../../packages/validate/src/lib/validation.ts:305:16)
  at RecordType.validate (../../../../../packages/store/src/lib/RecordType.ts:311:26)
  at StoreSchema.validateRecord (../../../../../packages/store/src/lib/StoreSchema.ts:381:22)
...
(30 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by ds300**
Closes #7997

When resizing a draw or highlight shape to zero width or height, `Box.Resize` returns `scaleX=0` or `scaleY=0`. The `onResize` handlers in `DrawShapeUtil` and `HighlightShapeUtil` multiplied this directly into the shape's `scaleX`/`scaleY` props, producing 0, which violates the `nonZeroFiniteNumber` validator and throws a `ValidationError`.

This PR guards against that by keeping the existing scale value when the computed result would be 0.

### Change type

- [x] `bugfix`

### Test plan

1. Create a draw shape on the canvas
2. Select it and resize by dragging an edge handle until the shape collapses to zero width or height
3. Verify no validation error is thrown

- [x] Unit tests

### Release notes

- Fixed a crash when resizing draw or highlight shapes to zero width or height.

Made with [Cursor](https://cursor.com)

---

> [!NOTE]
> **Low Risk**
> Small, localized guard in resize logic plus tests; behavior only changes for the edge case of collapsing a shape to zero size.
> 
> **Overview**
> Prevents `DrawShapeUtil` and `HighlightShapeUtil` from applying a resize operation that would result in `scaleX` or `scaleY` becoming `0`; the `onResize` handler now returns `undefined` (no-op) in that case.
> 
> Adds unit coverage for `DrawShapeUtil` to ensure resizing to zero width/height leaves scale unchanged and that negative scale (flipping) is preserved. Updates the generated API report to reflect the new `onResize` return type (`{ props: ... } | undefined`).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 2738894c90ea23acfeac45879432b11800b29565. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/tldraw/src/lib/shapes/draw/DrawShapeUtil.tsx [hunk 1]
```diff
@@ -229,6 +229,7 @@ export class DrawShapeUtil extends ShapeUtil<TLDrawShape> {
 
 	override onResize(shape: TLDrawShape, info: TLResizeInfo<TLDrawShape>) {
 		const { scaleX, scaleY } = info
+		if (scaleX === 0 || scaleY === 0) return
```
[the remark this answers is not in the document]
Comment (ds300) [reply packages/tldraw/src/lib/shapes/draw/DrawShapeUtil.tsx]: intentional
